pkg/cloudprovider/plugins/azure/azure_resourcegraph.go

```diff
@@ -55,9 +62,18 @@ func invokeResourceGraphQuery(resourceGraphAPIClient azureResourceGraphWrapper,
 		Facets:        nil,
 	}
 
-	results, queryErr := resourceGraphAPIClient.resources(context.Background(), request)
-	if queryErr == nil {
-		return results.Data, *results.TotalRecords, nil
+	// invoke resource graph API till all pages from response are fetched.
+	for currentRecords < totalRecords {
+		results, queryErr := resourceGraphAPIClient.resources(context.Background(), request)
+		if queryErr == nil {
+			data = append(data, results.Data)
+			currentRecords += *results.Count
```

Comment:
Can't you use check if the result.Count is less than pageSize, break. This will avoid totalRecord variable?

Reply:
I don't think we can use that, for result.Count exactly same as pageSize and totalRecord is also same as result.Count, we wouldn't want to enter the loop again. I think totalRecord is required.